src/orion/benchmark/task/task_wrapper.py

```diff
@@ -0,0 +1,183 @@
+        self.task: TaskType = task
+
+    def call(self, x: Any) -> List[Dict]:
+        return self.task.call(x)
+
+    def get_search_space(self) -> Dict[str, str]:
+        return self.task.get_search_space()
+
+    @property
+    def unwrapped(self) -> Union[TaskType, BaseTask]:
+        """ Returns the 'unwrapped' task. """
+        return self.task.unwrapped
+
+    @property
+    def configuration(self) -> Dict[str, Any]:
+        """Return the configuration of the task wrapper (including the wrapped task's configuration).
+        """
+        return {
+            type(self).__qualname__: {
+                "task": self.task.configuration,
+                "max_trials": self.max_trials,
+            }
+        }
+
+
+HParamsType = TypeVar("HParamsType")
+
+
+class FixTaskDimensionsWrapper(TaskWrapper[TaskType]):
+    """ Wrapper around a Task that fixes the values of some of its input dimensions.
+
+    The value returned by `get_search_space` is also modified.
+
+    Parameters
+    ----------
+    task : TaskType
+        A task to wrap.
+    fixed_dims : Dict[str, Any]
+        Dictionary mapping from the name of a dimension of the task space to the value that
+        dimension should be fixed at.
+    max_trials : int, optional
+        Maximum number of trials, by default None
+    """
+
+    def __init__(self, task: TaskType, fixed_dims: Dict[str, Any], max_trials: int = None):
+        super().__init__(task=task, max_trials=max_trials)
+        self.fixed_dims = fixed_dims
+        # The whole space, from the wrapped task.
+        self._full_space: Space = SpaceBuilder().build(self.task.get_search_space())
+
+        for dimension_name in fixed_dims:
+            if dimension_name not in self._full_space:
+                raise ValueError(
+                    f"Can't fix dimension '{dimension_name}' because it isn't in the task's space: "
+                    f"{self._full_space}."
+                )
+
+        # The part of the space that can be modified.
+        self._space: Space = SpaceBuilder().build(self.get_search_space())
+
+    def call(self, x: Any) -> List[Dict]:
+        """ Calls the wrapped task, adding/changing some dimensions in `x` if necessary. """
+        trial: Trial
+        if isinstance(x, np.ndarray):
+            point_tuple = regroup_dims(x, space=self._full_space)
+            trial = tuple_to_trial(point_tuple, space=self._full_space)
+        elif isinstance(x, dict):
+            trial = dict_to_trial(x, space=self._full_space)
+        elif isinstance(x, Trial):
+            trial = copy.deepcopy(trial)
+        else:
+            raise NotImplementedError(
+                f"Don't know how to fix some values for input {trial} of type {type(trial)}"
+            )
+
+        trial_was_modified = False
+        for name, fixed_value in self.fixed_dims.items():
+            params = trial.params
+
+            if name not in params:
```

Comment:
I'm copy-pasting a discussion here because we lost trace with the migration.

> @bouthilx bouthilx on 14 Sep 2021 Member
> Could we assume that X should be part of the subspace? The wrapped task defines the new subspace so the algorithms should only sample in that subspace.
> 
>  
> @lebrice lebrice on 14 Sep 2021 Author Contributor
> x is a part of the space of the space of the task wrapper, not necessarily a part of the space of the wrapped task.
> Let me know if that doesn't answer your question.
> 
>  
> @bouthilx bouthilx on 15 Sep 2021 Member
> Yeah so x is part of the subspace (without the fixed dim). That means only the code in the else clause is necessary.
> 
>  
> @lebrice lebrice on 21 Sep 2021 Author Contributor
> I see. But here I also handle the case where points are given that contain that dimension. I'll leave this as-is if that's ok with you.
> 
>  
> @bouthilx bouthilx on 12 Oct 2021 Member
> But is the task supposed to receive points with dimensions not contained in the search space? (They were removed by being fixed). If it's not needed removing it would be simpler for tests and maintenance.
> 
>  
> @lebrice lebrice on 6 Dec 2021 Author Contributor
> Having a dimension being fixed is not the same as not having that dimension. I don't understand this conversation anymore.
> This is a wontfix for me atm.
> 
>  
> @bouthilx bouthilx on 16 Dec 2021 Member
> I agree this should be moved to another PR since it is not used in this one.
> 
> What I was saying above is that for the experiment and algorithm this dimension does not exist, and therefore there should never be trials that transit between algo and experiment with this dimension. If we want to reuse trials from experiments where the dimension was not fixed, these trials should already be adapted by something like the EVC or some kind of knowledge base for warm-starting before they can reach the actual wrapped_task.__call__ method.
> 
>  
> @lebrice lebrice 15 days ago Author Contributor
> Fixed: removed this file.
>